modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/alert/AlertDefinitionManagerLocal.java

```diff
@@ -164,4 +164,18 @@ AlertDefinition updateAlertDefinitionInternal(Subject subject, int alertDefiniti
      */
     int removeResourceAlertDefinitions(Subject subject, int[] resourceAlertDefinitionIds);
 
+    /**
+     * Remove the specified alert definitions belongs to certain group. Currently deleted alert definitions are ignored,
+     * as are definitions for which the proper permission is not held.
+     * <p/>
+     * Requires MANAGE_ALERTS permission for the group or resource on which the alert definition is defined. REQUIRES
+     * MANAGE_SETTINGS for Templates.
+     *
+     * @param subject
+     * @param groupId
+     * @param alertDefinitionIds
+     * @return number of removed definitions
+     */
+    int removeAlertDefinitions(Subject subject,Integer groupId, int[] alertDefinitionIds);
```

Comment:
I suppose it would make sense to expose this as a public API, as is the only way to remove alert definitions using group permissions.

Reply:
I suppose we could keep this method variation under private API unless we are requested to expose it

//cc @spinder @loleary

Reply:
Although I would like it to be Public, I think you might be right about keeping it Private until someone explicitly requests it. Reasoning being:
 - JBoss ON is in maintenance mode.
 - This is really an enhancement.
 - The workaround of finding all resources that use the alert definition and manually removing it from said resources is doable via the API.
 - To expose it in the Public API, QE would need to define test cases and update test plans to include it for their remote API test suites.